polaris-core/src/main/java/org/apache/polaris/core/persistence/PolarisCredentialsBootstrap.java

```diff
@@ -63,6 +66,42 @@ public static PolarisCredentialsBootstrap fromString(@Nullable String credential
         : EMPTY;
   }
 
+  /**
+   * Parse a JSON array of credentials. Example: """ [ {"realm": "a", "principal": "root",
+   * "clientId": "abc123", "clientSecret": "xyz987"}, {"realm": "b", "principal": "boot",
+   * "clientId": "boot-id", "clientSecret": "boot-secret"}, ] """
+   */
+  public static PolarisCredentialsBootstrap fromJson(String json) {
+    ObjectMapper objectMapper = new ObjectMapper();
+    List<Map<String, String>> credentialsList = new ArrayList<>();
+    try {
+      credentialsList = objectMapper.readValue(json, new TypeReference<>() {});
+    } catch (Exception e) {
+      throw new IllegalArgumentException("Could not parse credentials JSON: " + json, e);
+    }
+    Map<String, Map.Entry<String, String>> credentials = new HashMap<>();
+    for (Map<String, String> entry : credentialsList) {
+      String realm = entry.get("realm");
+      String principal = entry.get("principal");
+      String clientId = entry.get("clientId");
+      String clientSecret = entry.get("clientSecret");
+      if (realm == null || principal == null || clientId == null || clientSecret == null) {
+        throw new IllegalArgumentException("Failed to find credentials in: " + json);
+      } else if (!principal.equals(PolarisEntityConstants.ROOT_PRINCIPAL_NAME)) {
+        throw new IllegalArgumentException(
+            String.format(
+                "Invalid principal %s. Expected %s.",
```

Comment:
Do we need the `principal` entry in the JSON at all?

Reply:
I can see how it's confusing, but I was imagining this as a sort of serialization of a `PolarisCredentials[Bootstrap]`... given that, including all the fields felt correct. We could also make `principal` optional.

Reply:
I think, in this case it would be fine to accept JSON with extra fields. However, in that case, the parsing code should probably not attempt to validate (or even interpret) the fields it does not use.

Reply:
I agree that extra fields are okay, but specifically I wanted to check for `principal` so as to not mislead people into thinking `principal` is a field that works. It must be `root` and I thought being explicit about that would be helpful.

Hopefully in the future we can lift this restriction and then we can simply remove that check without having to change the input syntax.

Reply:
Added a test to cover this behavior